sdk/storage/azblob/service/client.go

```diff
@@ -82,6 +83,23 @@ func NewClientFromConnectionString(connectionString string, options *ClientOptio
 	return NewClientWithNoCredential(parsed.ServiceURL, options)
 }
 
+// GetUserDelegationCredential obtains a UserDelegationKey object using the base ServiceURL object.
+// OAuth is required for this call, as well as any role that can delegate access to the storage account.
+func (s *Client) GetUserDelegationCredential(ctx context.Context, info KeyInfo, o *GetUserDelegationCredentialOptions) (*UserDelegationCredential, error) {
```

Comment:
I don't think we need this method. We DO need a method that returns a UDK but the customer can easily create a credential from this and so I see no reason to provide this additional functionality; we don't have it for any other kind of credential.